client/components/IndividualChallenge.js

```diff
@@ -1,22 +1,88 @@
-import React from 'react';
+/* eslint-disable react/no-unused-state */
+import React, { Component } from 'react';
+import CodeMirror from 'react-codemirror';
 
-export const IndividualChallenge = () => {
-  const challenge = { name: 'challenge1', description: 'draw a circle', difficulty: 1 };
-  return (
-    <div className="d-flex flex-column align-items-center">
-      <h1>{challenge.name}</h1>
-      <p>{challenge.description}</p>
-      <div>
-        <div className="row">
-          <div className="col">users page goes here</div>
-          <div className="col">our image goes here</div>
-        </div>
-        <div className="row">
-          <div className="col">HTML Editor</div>
-          <div className="col">CSS Editor</div>
-          <div className="col">JS Editor</div>
+export class IndividualChallenge extends Component {
+  constructor() {
+    super();
+    this.state = {
+      codeHTML: '',
+      codeCSS: '',
+      codeJS: '',
+    };
+    this.updateCodeHTML = this.updateCodeHTML.bind(this);
+    this.updateCodeCSS = this.updateCodeCSS.bind(this);
+    this.updateCodeJS = this.updateCodeJS.bind(this);
+    this.changeValue = this.changeValue.bind(this);
+  }
+
+  updateCodeHTML(newCode) {
+    this.setState({
+      codeHTML: newCode,
+    });
+  }
+
+  updateCodeCSS(newCode) {
+    this.setState({
+      codeCSS: newCode,
+    });
+  }
+
+  updateCodeJS(newCode) {
+    this.setState({
+      codeJS: newCode,
+    });
+  }
+
+  changeValue() {
+    // TODO: axios post call to the backend
+    console.log(this.state);
+  }
+
+  render() {
+    const challenge = { name: 'challenge1', description: 'draw a circle', difficulty: 1 };
+    const options = {
+      lineNumbers: true,
+      mode: 'javascript',
+    };
+    return (
+      <div className="d-flex flex-column align-items-center">
+        <h1>{challenge.name}</h1>
+        <p>{challenge.description}</p>
+        <div>
+          <div className="row">
+            <div className="col">users page goes here</div>
+            <div className="col">our image goes here</div>
+          </div>
+          <div className="row">
+            <div className="col">
+              <h4>HTML Editor</h4>
```

Comment:
Have an `h1`, so next is `h2`
```suggestion
              <h2>HTML Editor</h2>
```